plugins/radare2/radare2_main.ml

```diff
@@ -0,0 +1,62 @@
+open Bap_core_theory
+open Core_kernel
+open Bap_future.Std
+open Bap.Std
+include Self()
+
+open KB.Syntax
+
+let agent =
+  KB.Agent.register ~package:"bap.std" "radare2-symbolizer"
+    ~desc:"extracts symbols radare2"
+
+let provide_roots funcs =
+  let promise_property slot =
+    KB.promise slot @@ fun label ->
+    KB.collect Theory.Label.addr label >>| function
+    | None -> None
+    | Some addr ->
+      let addr = Bitvec.to_bigint addr in
+      Option.some_if (Hashtbl.mem funcs addr) true in
+  promise_property Theory.Label.is_valid;
+  promise_property Theory.Label.is_subroutine
+
+let provide_radare2 file = 
+  let funcs = Hashtbl.create (module struct
+      type t = Z.t
+      let compare = Z.compare and hash = Z.hash
+      let sexp_of_t x = Sexp.Atom (Z.to_string x)
+    end) in
+    let accept name addr = Hashtbl.set funcs addr name in
+    let extract name json = Yojson.Basic.Util.member name json in
+    try
+     let symbol_list = Yojson.Basic.Util.to_list (R2.with_command_j "aaa;aflj" file) in
```

Comment:
wouldn't be `aaa` here too much? 
@XVilka, what do you think, do we need triple a for symbols only?

Also, I think it would be a good idea to parameterize it by adding command-line options.

Reply:
It depends. If the elf file provides a symbolic information then its unnecessary. But if it doesn't, or its a raw binary - "aaa" is a way to go.

Reply:
@XVilka, but will `aaa` in a raw binary give the names? Symbolizer is used only for resolving names, e.g., to have `malloc` instead `sub_xxxx`.

Reply:
Ah, no.

Reply:
@XVilka, so the last questions,  what would be the minimal (not in terms of the number of characters, but in terms of the time of analysis) command that will give us names from radare2?

Reply:
When you say "names" are you referring to list of symbols? if so, the list can be found with the `is` command and wouldn't need analysis.

You need to check that it will indeed give you whatever you need :)

Reply:
`isj` then, or `aa;aflj` if you want basic analysis.

Reply:
> `isj` then, or `aa;aflj` if you want basic analysis.

Maybe a command-line option can be introduced here to make both of them available to user?

Reply:
Yes, I believe this would be the best solution, lets use `isj` as the default value and let the power-users to specify whatever command they would like.